server/src/main/java/org/elasticsearch/index/codec/vectors/es910/internal/hppc/HashContainers.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * @notice
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Modifications copyright (C) 2025 Elasticsearch B.V.
+ */
+
+package org.elasticsearch.index.codec.vectors.es910.internal.hppc;
+
+import org.apache.lucene.internal.hppc.BufferAllocationException;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.apache.lucene.util.BitUtil.nextHighestPowerOfTwo;
+
+/**
+ * Constants for primitive maps.
+ *
+ * @lucene.internal
+ */
+class HashContainers {
```

Comment:
I wonder if we can just extract what we want and directly place it into the fixed sized array thigns?